staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -105,6 +106,29 @@ func (c *cache) put(s string, a *Authenticator) *Authenticator {
 	return a
 }
 
+// returns the oldest certificate expiration time (oldest .NotBefore) among
+// all Authenticator objects within the cache. Useful for watching if any
+// certificates are nearing expiration. If cache contains no Authenticators
+// or no Authenticators contain a cached TLS certificate returns the zero time.
+func (c *cache) oldestExpiration() time.Time {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var oldest *time.Time
+	for _, a := range c.m {
+		if a.cachedCreds != nil &&
+			a.cachedCreds.cert != nil &&
+			a.cachedCreds.cert.Leaf != nil &&
+			(oldest == nil || a.cachedCreds.cert.Leaf.NotBefore.Before(*oldest)) {
+			oldest = &a.cachedCreds.cert.Leaf.NotBefore
+		}
+	}
+	if oldest == nil {
+		return time.Time{}
+	}
+	return *oldest
+}
+
```

Comment:
@liggitt @logicalhan 
This my solution for concurrent client certificates. We simply report the cert nearest to expiration since that is all we really care about.

Reply:
lgtm. would be nice to have a unit test around this, if you extract out lines 117-129, should be pretty easy to do.